frontend/src/main.ts

```diff
@@ -60,12 +60,12 @@ async function getFirebaseConfig(): Promise<Object> {
 
 getFirebaseConfig().then(config => {
   firebase.initializeApp(config);
-  firebase.auth().onAuthStateChanged(user => {
-    store.dispatch("user/handleUser", user);
+  firebase.auth().onAuthStateChanged(async user => {
+    store.dispatch("user/onAuthStateChanged", user);
   });
 
   const unsubscribe: any = firebase.auth().onAuthStateChanged(async user => {
-    await store.dispatch("user/handleUser", user);
+    await store.dispatch("user/handleInitialUserLoad", user);
```

Comment:
I guess we don't need to dispatch this action from this callback. The other callback above is already dispatching an action to the store to set the current user. Please think of another way to wait for the roles to be ready here.

Reply:
I have found that the roles do exist in state but the getter `hasRole` is not working as intended. With this knowledge I should have a better solution to this problem.